### PR TITLE
Added support for custom auth tokens

### PR DESCRIPTION
**Pull request by pontusmelke**

### CODE REVIEW

File: neo4j/v1/session.py [hunk 1]
```diff
@@ -510,14 +512,28 @@ def __ne__(self, other):
...
 
     :param user: user name
     :param password: current password
+    :param realm: specifies the authentication provider
+    :return: auth token for use with :meth:`GraphDatabase.driver`
+    """
+    return AuthToken("basic", user, password, realm)
+
+
+def custom_auth(principal, credentials, realm=None, scheme=None, parameters=None):
```
Comment (craigtaverner) [inline neo4j/v1/session.py]: Do we want optional params for realm and scheme? Not consistent with java driver.

File: neo4j/v1/session.py [hunk 2]
```diff
@@ -49,10 +47,14 @@ class AuthToken(object):
     """ Container for auth information
     """
 
-    def __init__(self, scheme, principal, credentials):
+    def __init__(self, scheme, principal, credentials, realm=None, parameters=None):
```
Comment (technige) [inline neo4j/v1/session.py]: Should pass parameters as kwargs.

File: test/test_session.py [hunk 3]
```diff
@@ -97,6 +97,29 @@ def test_fail_nicely_when_connecting_to_http_port(self):
         assert str(context.exception) == "Server responded HTTP. Make sure you are not trying to connect to the http " \
                                     "endpoint (HTTP defaults to port 7474 whereas BOLT defaults to port 7687)"
 
+    def test_can_provide_real_with_basic_auth_token(self):
```
Comment (zhenlineo) [inline test/test_session.py]: real -> realm?

**Comment by technige** [comment]
Replaced by #103

**Pull request closed without merging**